backend/models/upload.js

```diff
@@ -0,0 +1,240 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+"use strict";
+
+const fs = require("fs");
+const multer = require("multer");
+const nodeuuid = require("uuid/v1");
+const config = require("../config");
+const C = require("../constants");
+const utils = require("../utils");
+const middlewares = require("../middlewares/middlewares");
+const responseCodes = require("../response_codes");
+const systemLogger = require("../logger.js").systemLogger;
+const importQueue = require("../services/queue");
+const { modelStatusChanged } = require("./chatEvent");
+const { isValidTag } = require("./history");
+const { findModelSettingById, setCorrelationId } = require("./modelSetting");
+
+const checkFileFormat = async (filename) => {
+	let format = filename.split(".");
+
+	if (format.length <= 1) {
+		throw responseCodes.FILE_NO_EXT;
+	}
+
+	const isIdgn = format[format.length - 1] === "dgn" && format[format.length - 2] === "i";
+
+	format = format[format.length - 1];
+
+	if (isIdgn || C.ACCEPTED_FILE_FORMATS.indexOf(format.toLowerCase()) === -1) {
+		throw responseCodes.FILE_FORMAT_NOT_SUPPORTED;
+	}
+};
+
+const handleChunkStream = async (req, filename) => {
+	return new Promise(resolve => {
+		const writeStream = fs.createWriteStream(filename, {encoding: "binary"});
+		req.on("data", (chunk) => writeStream.write(chunk, "binary"));
+		req.on("end", () => {
+			writeStream.end();
+			resolve();
+		});
+	});
+};
+
+const stitchChunks = (corID, newFilename) => {
+	const filePath = `${importQueue.getTaskPath(corID)}/chunks`;
+
+	const files = fs.readdirSync(filePath);
+
+	// sort timestamps into ascending order
+	files.sort((a, b) => a - b);
+
+	files.forEach((file) => {
+		// for debugging
+		const stats = fs.statSync(`${filePath}/${file}`);
+		systemLogger.logInfo(`FILE=${filePath}/${file}, SIZE=${stats.size} bytes`);
+		// end debugging
+
+		const content = fs.readFileSync(`${filePath}/${file}`);
+
+		fs.appendFileSync(`${importQueue.getTaskPath(corID)}/${newFilename}`, content);
+	});
+};
+
+const Upload = {};
+
+Upload.initChunking = async (teamspace, model, username, data) => {
+	if (!data.filename) {
+		throw responseCodes.INVALID_ARGUMENTS;
+	}
+
+	await checkFileFormat(data.filename);
+
+	// check model exists before upload
+	const modelSetting = await findModelSettingById(teamspace, model);
+
+	if (!modelSetting) {
+		throw responseCodes.MODEL_NOT_FOUND;
+	}
+
+	const newFileName = data.filename.replace(C.FILENAME_REGEXP, "_");
+
+	await isValidTag(teamspace, model, data.tag);
+
+	const corID = nodeuuid();
+
+	await importQueue.writeImportData(
+		corID,
+		teamspace,
+		model,
+		username,
+		newFileName,
+		data.tag,
+		data.desc,
+		data.importAnimations
+	);
+
+	return { corID };
+};
+
+Upload.uploadFile = async (req) => {
+	if (!config.cn_queue) {
+		return Promise.reject(responseCodes.QUEUE_NO_CONFIG);
+	}
+
+	const { account, model } = req.params;
```

Comment:
could we refactor this so anything request/multer related stays in route folder. Where possible we want to avoid request data bleeding into the models folder. route handles requests, model handles internal logic.

I think majority of the logic here should just be in `routes`